features/storage/kvstore/conf/kv_config.cpp

```diff
@@ -0,0 +1,837 @@
+static bool is_kv_config_initialize = false;
+static kvstore_config_t kvstore_config;
+
+#define INTERNAL_BLOCKDEVICE_NAME FLASHIAP
+
+#define STR_EXPAND(tok) #tok
+#define STR(tok) STR_EXPAND(tok)
+
+#define _GET_FILESYSTEM_concat(dev, ...) _get_filesystem_##dev(__VA_ARGS__)
+#define GET_FILESYSTEM(dev, ...) _GET_FILESYSTEM_concat(dev, __VA_ARGS__)
+
+#define _GET_BLOCKDEVICE_concat(dev, ...) _get_blockdevice_##dev(__VA_ARGS__)
+#define GET_BLOCKDEVICE(dev, ...) _GET_BLOCKDEVICE_concat(dev, __VA_ARGS__)
+
+static inline uint32_t align_up(uint64_t val, uint64_t size)
+{
+    return (((val - 1) / size) + 1) * size;
+}
+
+static inline uint32_t align_down(uint64_t val, uint64_t size)
+{
+    return (((val) / size)) * size;
+}
+
+int _calculate_blocksize_match_TDBStore(BlockDevice *bd)
+{
+    bd_size_t size = bd->size();
+    bd_size_t erase_size = bd->get_erase_size();
+    bd_size_t number_of_sector = size / erase_size;
+
+    if (number_of_sector < 2){
+        tr_warning("KV Config: there is less then 2 sector TDBStore will not work.");
+        return -1;
+    }
+
+
+    if (number_of_sector % 2 != 0){
+        tr_warning("KV Config: Number of sector is not even number. Consider changing the BlockDevice size");
+    }
+
+    return MBED_SUCCESS;
+}
+
+int _get_addresses(BlockDevice *bd, bd_addr_t start_address, bd_size_t size, bd_addr_t *out_start_addr,
+                   bd_addr_t *out_end_addr )
+{
+    bd_addr_t aligned_end_address;
+    bd_addr_t end_address;
+    bd_addr_t aligned_start_address;
+
+    aligned_start_address = align_down(start_address, bd->get_erase_size(start_address));
+    if (aligned_start_address != start_address) {
+        tr_error("KV Config: Start address is not aligned. Better use %02llx", aligned_start_address);
+        return -1;
+    }
+
+    if (size == 0) {
+        (*out_start_addr) = aligned_start_address;
+        (*out_end_addr) = bd->size();
+        return 0;
+    }
+
+    end_address = start_address + size;
+    aligned_end_address = align_up(end_address, bd->get_erase_size(end_address));
+    if (aligned_end_address != end_address) {
+        tr_error("KV Config: End address is not aligned. Consider changing the size parameter.");
+        return -1;
+    }
+
+    if (aligned_end_address > bd->size()) {
+        tr_error("KV Config: End address is out of boundaries");
+        return -1;
+    }
+
+    (*out_start_addr) = aligned_start_address;
+    (*out_end_addr) = aligned_end_address;
+    return 0;
+}
+
+FileSystem *_get_filesystem_FAT(BlockDevice *bd, const char *mount)
+{
+    static FATFileSystem sdcard(mount, bd);
+    return &sdcard;
+
+}
+
+FileSystem *_get_filesystem_LITTLE(BlockDevice *bd, const char *mount)
+{
+    static LittleFileSystem flash(mount, bd);
+    return &flash;
+}
+
+FileSystemStore *_get_fileSystemStore(FileSystem *fs)
+{
+    static FileSystemStore fss(fs);
+    return &fss;
+}
+
+FileSystem *_get_filesystem_default(BlockDevice *bd, const char *mount)
```

Comment:
`_get_filesystem_default` vs `_get_fileSystemStore` - I would follow the former (no camelcase for function names). SImilar in other functions in this file, like `_calculate_blocksize_match_TDBStore` -> `_calculate_blocksize_match_tbdstore`. or I even noticed `_storage_config_TDB_EXTERNAL`, please consolidate

(this might be normal in the storage files?)

Reply:
The following cannot be changed cause the calling function will build the correct function name using configuration macro:
_get_filesystem_FAT
_get_filesystem_LITTLE
_get_blockdevice_FLASHIAP
_get_blockdevice_SPIF
_get_blockdevice_QSPIF
_get_blockdevice_DATAFLASH
_get_blockdevice_SD
_storage_config_TDB_INTERNAL
_storage_config_TDB_EXTERNAL
_storage_config_TDB_EXTERNAL_NO_RBP
_storage_config_FILESYSTEM
_storage_config_FILESYSTEM_NO_RBP

the rest I have changed to lower case.

Reply:
That one I have missed but noticed some string macros. How does this work (this function_MACRO) here?

Reply:
Yes, by design the mbed_lib.json value of default is in lower case.

Reply:
Done.